### PR TITLE
remove the use of File::HomeDir

### PR DESCRIPTION
**Pull request by karenetheridge**
..as this pulls in extra dependencies on darwin, which complicates static deployment files.
Also removed (nearby) dangerous uses of "my ... = ... or ..." which can result in inadvertent closures.

### CODE REVIEW

**Comment by theory** [comment]
Well, is there not a better module for this, then?

**Comment by theory** [comment]
Also, can you change it to merge into the develop branch?

**Comment by karenetheridge** [comment]
I have rebased against develop. I would suggest you update the settings for the repository so that clones automatically base from develop rather than master, to avoid future confusion.

There is File::HomeDir::Tiny, but it has an outstanding bug, resulting in windows malfunctions: https://rt.cpan.org/Ticket/Display.html?id=127425

**Comment by theory** [comment]
Yeah, @ssoriche and I tweaked the branch settings this afternoon. Lesson learned.

File: lib/App/Sqitch/Config.pm [hunk 1]
```diff
@@ -21,10 +21,8 @@ has '+encoding' => ( default => 'UTF-8' );
 my $SYSTEM_DIR = undef;
 
 sub user_dir {
-    require File::HomeDir;
-    my $hd = File::HomeDir->my_home or hurl config => __(
-        "Could not determine home directory"
-    );
+    my $hd = $^O eq 'MSWin32' && "$]" < '5.016' ? $ENV{HOME} || $ENV{USERPROFILE} : (glob('~'))[0];
```
Comment (theory) [inline lib/App/Sqitch/Config.pm]: Should probably use App::Sqitch::ISWIN where appropriate.

**Comment by karenetheridge** [comment]
Thanks!

**Pull request merged**